assembl/static2/js/app/components/debate/survey/question.jsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+import { graphql } from 'react-apollo';
+import gql from 'graphql-tag';
+import { connect } from 'react-redux';
+import { Translate, I18n } from 'react-redux-i18n';
+import { Grid, Col, FormGroup, FormControl, Button } from 'react-bootstrap';
+import { getIfPhaseCompletedByIdentifier } from '../../../utils/timeline';
+
+class Question extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showSubmitButton: false
+    };
+    this.updateDimensions = this.updateDimensions.bind(this);
+    this.getProposalText = this.getProposalText.bind(this);
+    this.createPost = this.createPost.bind(this);
+  }
+  componentDidMount() {
+    const maxChars = this.txtarea.props.maxLength;
+    this.state = {
+      remainingChars: maxChars
+    };
+    this.updateDimensions();
+    window.addEventListener('resize', this.updateDimensions);
+  }
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.updateDimensions);
+  }
+  getRemainingChars(e) {
+    const maxChars = this.txtarea.props.maxLength;
+    const remainingChars = maxChars - e.currentTarget.value.length;
+    this.setState({
+      remainingChars: remainingChars
+    });
+  }
+  getProposalText(e) {
+    const txtValue = e.currentTarget.value;
+    this.setState({
+      postBody: txtValue
+    });
+  }
+  displaySubmitButton(e) {
+    const nbChars = e.currentTarget.value.length;
+    if (nbChars > 10) {
+      this.setState({
+        showSubmitButton: true
+      });
+    } else {
+      this.setState({
+        showSubmitButton: false
+      });
+    }
+  }
+  updateDimensions() {
+    const componentHeight = this.question.clientHeight;
+    const screenHeight = window.innerHeight - document.getElementById('navbar').clientHeight;
+    const screenWidth = window.innerWidth;
+    setTimeout(() => {
```

Comment:
I don't know why I did this. I remove it because it's not necessary to do a setTimeout. Good point again!